### ISSUE TITLE
[Enhancement] Some more parsers

### ISSUE THREAD

**Issue by typhoon71** [dteviot/WebToEpub#71]
- [ ] https://www.epub.pub/
- [ ] https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc
- [x] http://www.69shu.com/ 
- [x] unlimitednovelfailures.mangamatters.com
- [x] https://creativenovels.com/ 
- [x] dreamsleeper87.blogspot.sg/p/world-customize-creator-web-novel.html
- [x] https://fantasy-books.live/
- [x] frostfire10.wordpress.com
- [x] gravitytales.com
- [x] hellping.org
- [x] isekaicyborg.wordpress.com
- [x] japtem.com
- [x] krytykal.org
- [x] http://liberspark.com
- [x] lightnovelbastion.com
- [x] lnmtl.com
- [x] moonbunnycafe.com
- [x] nanodesutranslations.wordpress.com / www.*thetranslation.wordpress.com 
- [x] http://www.novelall.com
- [x] http://novelfull.com/
- [x] http://novelonlinefree.com
- [x] https://www.noveluniverse.com/
- [x] www.pathoftranslation.com
- [x] www.radianttranslations.com
- [x] shalvationtranslations.wordpress.com
- [x] shikkakutranslations.org
- [x] skythewood.blogspot.it
- [x] www.sousetsuka.com
- [x] www.translationnations.com
- [x] volaretranslations.com
- [x] https://www.webnovel.com/ aka https://en.qidian.com/
- [x] https://wnmtl.com
- [x] http://www.wuxiaworld.co/
- [x] *.wikipedia.org
- [x] www.xianxiaworld.net
- [x] yoraikun.wordpress.com
- [x] http://zenithnovels.com

Since I can't do them myself, I'll shamelessly ask for some new parsers to be added.

moonbunnycafe.com -> This is a site where novel translations are hosted with TLs "autorization", and there are a bunch of nice ones; it would make a good addiction.

nanodesutranslations.wordpress.com, or better www.*thetranslation.wordpress.com (note the "*")
-> TL, I think this was already asked talked about, but I'll put here since there's a lot of content there.

krytykal.org -> TL, good stuff
unlimitednovelfailures.mangamatters.com -> TL, good stuff (but slow updates)

There are lot of other sites I'd like to ask, but I'll refrain since they mostly are 1 project TL each site and it would be a lot of work for little gain.

/me reloads shame concept
/me blushes

**Comment by AzSiAz** [comment]
krytykal.org > it seem easy every chapter is inside a div with ```su-posts su-posts-single-post``` as a css class.

unlimitednovelfailures.mangamatters.com > Some weird html here new chapter title is inside previous chapter tag but chapter content is in the new ```<p>``` tag, should be somewhat easy

nanodesutranslations > Might be tricky since it's not the same person who post it but a classic wordpress parser should be able to handle it chapter by chapter, except somme weird case. I don't remember them doing full text no ?

moonbunnycafe.com > Same parser from nanodesutranslations should do it chapter by chapter

**Comment by dteviot** [comment]
@AzSiAz 

> I remember seeing something to parse provided chapter url and make a volume but I don't see it
Did I dream about this ?

You're not far off.  I'm currently trying to make that logic more general.
Have a look at the ZirusMusing and WuxiaWorld parsers for the basic idea.

**Comment by dteviot** [comment]
@typhoon71 

> There are lot of other sites I'd like to ask, but I'll refrain since they mostly are 1 project TL each site and it would be a lot of work for little gain.

Please list them.  If I'm lucky, it may be possible to generalize and get many of them with a single parser.  
I'm currently thinking about a general "wordpress" parser.

**Comment by typhoon71** [comment]
Oh, my post is editing itself! XD

btw, I noticed ZirusMusing content is now hosted on moonbunnycafe.
Nice to know you're making the addon more powerful.
I'll (soon) add some site on the list that appeared in my post.

**Comment by AzSiAz** [comment]
@dteviot 
Oh right I will get a closer look at those parser thanks :)
I think it would be useful for a general wordpress parser since most of them doesn't have a full text page

**Comment by dteviot** [comment]
@typhoon71  @AzSiAz 
Added some new parsers. On the ExperimentalTabMode branch.
Note, I've only done a quick test on one or two stories per site.

- frostfire10.wordpress.com: need to go to web page with list of chapters for it to work. (https://frostfire10.wordpress.com/chapters/)
- isekaicyborg.wordpress.com: need to go to page with list of chapters, then select the chapters to pack.
- shalvationtranslations.wordpress.com: need to go to page with list of chapters, then select the chapters to pack.
- hellping.org:   parser shows links for all chapters of all stories, so you need to pick the chapters to pack carefully.
- krytykal.org: parser shows links for all chapters of all stories, so you need to pick the chapters to pack carefully.

**Comment by typhoon71** [comment]
1) frostfire10.wordpress.com -> OK, works
2) isekaicyborg.wordpress.com -> OK, works, but:
-spurious link (http://cont.lanove.kodansha.co.jp)
-needs reloading after getting some chapters (like, after getting v1 to getv2) if OS is suspended.
3) shalvationtranslations.wordpress.com -> OK, works, but:
-spurious links (http://www.mediafire.com, https://drive.google.com)
-I can't find the option to remove duplicates, the images in the gallery contain some that are later shown in the novel
-can't choose the cover image from the gallery
-default cover image is not resized/centered (well, ok, it's small, but still)

4) hellping.org -> OK, works
5) krytykal.org -> OK, works, but:
-I can't find the option to remove duplicates, the images in the gallery contain some that are later shown in the novel
-can't choose the cover image from the gallery
-default cover image is not resized/centered (well, ok, it's small, but still)

Regarding "parser shows links for all chapters of all stories" for 4) & 5) , it's easy enough to check the right ones, the link suggests what is what; seems clear and easy enough for me.
I did not test ALL of the novels, mostly the ones I like.

**Comment by dteviot** [comment]
@typhoon71

> 2) isekaicyborg.wordpress.com -> OK, works, but:
-spurious link (http://cont.lanove.kodansha.co.jp)
-needs reloading after getting some chapters (like, after getting v1 to getv2) if OS is suspended.
3) shalvationtranslations.wordpress.com -> OK, works, but:
-spurious links (http://www.mediafire.com, https://drive.google.com)

That's kind of by design.  Currently, it's just grabbing all hyperlinks, and I leave it to the user to remove any that are not wanted. 

> I can't find the option to remove duplicates, the images in the gallery contain some that are later shown in the novel

There isn't one.  It's currently only supported for Baka-Tsuki, because I can identify the image gallery there.  I can't do that in the general case.  That said, might be possible to make an educated guess, if a page contains mostly images.

>can't choose the cover image from the gallery

You should be able to supply the URL of the image you want used as cover.

>default cover image is not resized/centered (well, ok, it's small, but still)

Odd.  It should be.  I've wrapped it in a <svg> <image>.

**Comment by typhoon71** [comment]
With "can't choose the cover image from the gallery" I meant that they doesn't get shown like with baka-tsuki, where you just need to select "use as cover". 
That said, I am indeed able to supply the URL of the image I want used as cover.

About default cover, I was wrong, sorry: it works as intended.
The thing is I also open the epubs with sumatra to check for if the image link option works, and with it the cover image is like I said; it's sumatra fault, I suppose it's stripping formatting / style... Caligre show the image correctly.

**Comment by dteviot** [comment]
@typhoon71
> With "can't choose the cover image from the gallery" I meant that they doesn't get shown like with baka-tsuki, where you just need to select "use as cover".

That's because with Baka-Tsuki, the whole story including images is on a single page, so the images can be shown at the start.  With all the others, the images are known only after loading all the pages.  

This has implications for removing duplicate images from gallery.  Usually the gallery is at the start, but the images that are duplicated elsewhere are not known until all the rest of the pages are loaded.

I'm thinking that a solution to this problem is move the galleries to the end of the book, rather than have them at the start.  Of course, if this is done, then there may be no need to remove the duplicates as the risk of spoilers is reduced.

What do you think?

**Comment by typhoon71** [comment]
mmm, kind of hard to answer.

I like the gallery being at the start, whithout duplicates, but not because of spoilers: it's just because it's more similar to the actual "book". Also most epub of novels have the gallery at the start.
My choice would be to put the gallery at the start, even with duplicates if it's not possible to remove them.

But I do recognize having the gallery at the end to reduce the risk of spoilers is a valid point.

So... An option for the user between those two? Like "put the gallery at the end if it's not bakatsuki"?

**Comment by dteviot** [comment]
@typhoon71

Added basic support for
- GravityTales
- moonbunnycafe.com
- nanodesutranslations.wordpress.com  (Note, go to one of the  www.*thetranslation.wordpress.com pages and run extension.)
-  japtem.com

toshiya44
 You suggested Nanodesu and hellping, so you might like to give them a go.

**Comment by dteviot** [comment]
@typhoon71

Added basic support for
- GravityTales
- moonbunnycafe.com
- nanodesutranslations.wordpress.com  (Note, go to one of the  www.*thetranslation.wordpress.com pages and run extension.)
-  japtem.com

@toshiya44
 You suggested Nanodesu and hellping, so you might like to give them a go.

**Comment by typhoon71** [comment]
**GravityTales**
Tested with "The Experimental Log of the Crazy Lich": the title disappears from the chapter text. 

**moonbunnycafe.com**
Tested with "The Guild’s Cheat Receptionist" and "Lazy Dungeon Master", seems fine.

**japtem.com**
Tested with "Magi's Grandson" Volume 2: the image gallery is messed up (missing images, duplicate/wrong/ issing images); some spurious text before the chapter ("this chapter was brough to you by....")

**Nanodesu** 
Tested with "Fire Girl" and "Sasami-san Gambaranai": spurious "next page before the chapter start and before and after the notes for "Fire Girl", triple separators for both sometime. 
Here a screenshot of a page where both happen (Fire Girl): 

![fgsample](https://cloud.githubusercontent.com/assets/1866248/18408591/40f552b6-7734-11e6-8062-31f655efbc1e.png)



Did not test nanodesu much because nanodesu provides pdf/epub after the volume are done and edited (takes time but I'm not in a hurry)

**Comment by dteviot** [comment]
@typhoon71
>GravityTales
Tested with "The Experimental Log of the Crazy Lich": the title disappears from the chapter text.

Fixed.

>japtem.com
Tested with "Magi's Grandson" Volume 2: the image gallery is messed up (missing images, duplicate/wrong/ issing images); 

I think the missing/duplicate image is an issue with the page itself. http://japtem.com/projects/magis-grandson-toc/v2_illustrations/
The list of all images is missing image 5 from the slide show, and instead duplicates the 4th image in the show.
Note, at current time, parser doesn't vist the slideshow on imagur.  It uses the images in the "All Images" section.

> some spurious text before the chapter ("this chapter was brough to you by....")

I'll look into that.

> Nanodesu
Will investigate.

Have added skythewood.blogspot.it to ExperimentalTabMode

**Comment by typhoon71** [comment]
**GravityTales**
The chapter title is back, but now there's some unwanted text at the start and at the end of the chapter, like "[Previous Chapter][Table of Content][Next Page]"...

**japtem.com**
Yeah, I noticed the addon was using the gallery (which has wrong images) instead of navigating imgur (where there are the correct images); I suppose they should fix the webpage, but it wouldn't hurt to consider the case of galleries both on local and imgur.

**skythewood.blogspot.it** 
Tested with "Altina the sword princess": I noticed that the font (size) of the prologue text and the one of ch1 are different while I'd expect them to be the same; the web pages are like that, so I don't think it's the addon doing that. Would it be possible to have the same font size?

**Comment by dteviot** [comment]
@typhoon71 
> japtem.com
some spurious text before the chapter ("this chapter was brough to you by....")

That's part of the story text, it's not marked in any way.

>GravityTales
The chapter title is back, but now there's some unwanted text at the start and at the end of the chapter, like "[Previous Chapter][Table of Content][Next Page]"...

The problem here is that the links are using different URLs than given in the ToC.  So parser doesn't know that they're next/previous.  e.g. 
- In ToC, the first chapters is http://gravitytales.com/the-experimental-log-of-the-crazy-lich/elcl-chapter-1/, 
- but on Chapter 2, the link to chapter 1 is http://gravitytales.com/elcl-chapter-1/

**Comment by dteviot** [comment]
@typhoon71
shikkakutranslations.org should now be working if you want to give it a try.

**Comment by typhoon71** [comment]
**shikkakutranslations.org**
Tested with Kamigoroshi, seems to work perfectly. Thanks.

Since both of the novels translated on this site appear in the chapter list, it takes a lot to remove the unwanted ones (100+): would it be possible to implement something like intervals of chapters to select/unselect? Should I add this request to #73?

**Comment by dteviot** [comment]
@typhoon71
>Should I add this request to #73?

Yes please.  I'm currently thinking of opening a multi-line text editor, with a hyperlink entry on each line.
The idea is someone can then add, delete, edit, re-arrange, etc the links.  i.e. There's a button like "edit chapter links".  This replaces the table of links with the text editor.  User can then re-arrange the links as  I've described, then press the "links as table" to convert back to a table.  
This is easiest for me to implement and most flexible for a user, but does require the user to have a minimal knowledge of hyperlinks.  Also, there's the possibility that they could mistype something i.e. Not correctly escape a special character.
What do you think?

**Comment by typhoon71** [comment]
Seems good to me, flexible and not too complicated.
Btw, adding suggestion to #73.

**Comment by dteviot** [comment]
@typhoon71 
> skythewood.blogspot.it 
Tested with "Altina the sword princess": I noticed that the font (size) of the prologue text and the one of ch1 are different while I'd expect them to be the same; the web pages are like that, so I don't think it's the addon doing that. Would it be possible to have the same font size?

Have raised new issue.  https://github.com/dteviot/WebToEpub/issues/75

**Comment by typhoon71** [comment]
"http://www.readlightnovel.com/modern-weapons-cheat-in-another-world": the title disappears from the chapter text (same thing that was happening on GravityTales.

**Comment by dteviot** [comment]
@typhoon71
>"http://www.readlightnovel.com/modern-weapons-cheat-in-another-world": the title disappears from the chapter text (same thing that was happening on GravityTales.

I'm not too worried about this just now.  Not all stories on readlightnovel are formatted the same.
(For example, AccelWorld is encrypted/scrambled to make it difficult to parse.)



Have added a "default parser" that is used when plug-in can't identify parser to use to ExperimentalTab branch.
Parser is very crude.  It looks for the element you tell it to look for:
Either &lt;body&gt;, &lt;div&gt; or &lt;article&gt;
You can specify the &lt;div&gt; and &lt;article&gt; tags slightly better.  Options are
- First Element of that type.
- Element with specified class or id.
- Elelement with class or id that starts with specified string.

It defaults to grabbing the entire &lt;body&gt; element.

Please take it for a spin.
Caution: I'm not sure how good the error handling is if you tell it to look for a content element that is not present.

It works for unlimitednovelfailures.mangamatters.com if you set the element to look for to &lt;div&gt; "class starts with entry"
Note, this site has entire stories as single page. The default parser will not split a page into chapters.

**Comment by dteviot** [comment]
@typhoon71
> I'm not too worried about this just now.  Not all stories on readlightnovel are formatted the same.
(For example, AccelWorld is encrypted/scrambled to make it difficult to parse.)

Well, this is embarrassing.  Looks like I never added the chapter titles because I didn't think they added anything.
They were just "EPUB Name" - Volume X - Chapter Y.
However, on the basis you believe this is useful, title is now added to each chapter.

**Comment by toshiya44** [comment]
Sometimes the full size image is not downloaded, for example https://bibliathetranslation.wordpress.com/volume-2/chapter-1/
In the epub I saw,
```
    <img src="../Images/0004_00004.jpg" alt=""/>
    <!--  https://bibliathetranslation.files.wordpress.com/2016/01/00004.jpg?w=774  -->
```
Same for the image at https://bibliathetranslation.wordpress.com/volume-2/
However the images at https://bibliathetranslation.wordpress.com/volume-2/color-art/ were downloaded at full size.

**Comment by typhoon71** [comment]
>  However, on the basis you believe this is useful, title is now added to each chapter.
Thanks, it works. I do find them helpful, even if I have to edit them a bit (sometime, np).

> unlimitednovelfailures.mangamatters.com
This site loads a list of link, but doesn't work and doesn't pack, complaining about a html when it expected an image (or more like I don't get how to set the stuf you talked about).
The site "works" like this: one link is a full volume (complete or ongoing), so for example "http://unlimitednovelfailures.mangamatters.com/risou-no-himo-seikatsu/risou-no-himo-seikatsu-volume-03/" contains the whole volume 3 with pics. The post just point to the correct position inside the html. Also in that html there's a TOC in form of liks at the start. I don't think it's standard?

Will have to test more the general parser later on.

**Comment by dteviot** [comment]
@toshiya44  
>Sometimes the full size image is not downloaded, for example https://bibliathetranslation.wordpress.com/volume-2/chapter-1/ 

That's because of the information in the file.
For the above file, the only image information is this:
```html
<img class="alignnone size-full wp-image-481" src="https://bibliathetranslation.files.wordpress.com/2016/01/00004.jpg?w=774" alt="00004">
````

For the the full size images the image information is 

```html
<a href="https://bibliathetranslation.files.wordpress.com/2016/01/00001.jpg" rel="attachment wp-att-479">
    <img class="alignnone size-full wp-image-479" src="https://bibliathetranslation.files.wordpress.com/2016/01/00001.jpg?w=774&amp;h=1106" alt="00001" width="774" height="1106">
</a>
```

Note that in this case the &lt;img&gt; is enclosed in a &lt;a&gt;.  When the parser sees this sort of thing, it will try to fetch the &lt;a&gt; as the full size image.
What I should probably do is remove the "w" and "h" query values from the url.  This is on my todo list.  Refer https://github.com/dteviot/WebToEpub/issues/74


>Also, the TN notes are broken, but I guess there's no easy solution for that... 

Can you tell me what you mean in more detail please?
Looking at https://bibliathetranslation.wordpress.com/volume-1/chapter-3-vol-1/ there are no hyperlinks between the footnotes and the text they refer to.

@typhoon71
> complaining about a html when it expected an image 

OK, that's related to the issue I described above to toshiya, when an image is wrapped in a hyperlink, the parser assumes the link is to the full size image.  In this case, it's not a link to the full size image, but to the main web site.  Note that when this happens, it will not (or it should not) prevent the page being packed into an EPUB.

> I don't get how to set the stuf you talked about 

On the popup dialog, 6th line from top, just above the “pack epub” button and below the “Filename” control.
There's a line starting with “Element with Chapter Content”, then a drop down with &lt;body&gt;, &lt;article&gt;, &lt;div&gt;, then a drop down with “First Found”, etc. then a text box.

You use these to specify the element that holds the content for a page.  For each page in the links, the parser will fetch the page, then extract the element you specify with the above settings.
As you didn't change the settings, it will have grabbed the &lt;body&gt; element.  i.e. pretty much the entire page.  (Side note, obviously these settings are not sufficiently obvious. I guess I'll need to have them on a separate dialog as part of the open sequence.)

This is probably why you got the warning about the image.  There is an image that links to the main site on the page, but it's outside the story content element.  i.e. this
```html
<a href="http://unlimitednovelfailures.mangamatters.com/">					<img src="http://unlimitednovelfailures.mangamatters.com/wp-content/uploads/2013/06/011.gif" alt="Unlimited Novel Failures" id="logo"/>
				</a>
```


If you'd extracted the story content element, you would have skipped that link.

> but doesn't work and doesn't pack, 

Odd, it works for me.
My steps.
- In chrome browse to http://unlimitednovelfailures.mangamatters.com/risou-no-himo-seikatsu/risou-no-himo-seikatsu-volume-03/
- Click on WebToEpub to open the tab.
- Click OK on the warning.
- Set first drop down to &lt;div&gt;
- Set second drop down to “Class starts with”
- Set text to “entry”  (ignore the quotes)
- Set “Cover Image URL:” to http://unlimitednovelfailures.mangamatters.com/wp-content/uploads/2015/05/img004b.jpg
- Unselect all links except for http://unlimitednovelfailures.mangamatters.com/risou-no-himo-seikatsu/risou-no-himo-seikatsu-volume-03/
- Click “Pack epub”

>The site "works" like this: one link is a full volume (complete or ongoing), so for example "http://unlimitednovelfailures.mangamatters.com/risou-no-himo-seikatsu/risou-no-himo-seikatsu-volume-03/" contains the whole volume 3 with pics. 

Yes, I realize that.

> The post just point to the correct position inside the html. Also in that html there's a TOC in form of liks at the start. I don't think it's standard? 

OK, I think I see at least part of the confusion.
The default parser is very simple, it just grabs each link it's given, extracts the specified content element and writes it as a “chapter” to the EPUB.
So
- You need to prune the set of links you give so that there's only 1 per HTML page you want to fetch.  i.e. you need to ignore the links that are to the actual chapters on the page
- The parser will fetch the entire volume as a single chapter.  You will need to split it (and remove anything you don't want.)  However, it will fetch the images and adjust the links.

**Comment by typhoon71** [comment]
OK, I've seen the light!
I had to clean the cache to have the dropdown show itself: now I do get it... and I can pack the epub(s), yay! It seems to be working fine here too.
BTW, could you add a checkbox to disable the cover image custom link usage? Easier than deleting. ;)

**Comment by toshiya44** [comment]
On this page, there's a reference link, but it's formatted weirdly. 
And it appears like this in the epub:

![ref](https://cloud.githubusercontent.com/assets/20046789/18818600/7df36b12-834b-11e6-8379-b8545b1e558b.PNG)

**Comment by dteviot** [comment]
@toshiya44 

> Sometimes the full size image is not downloaded, for example https://bibliathetranslation.wordpress.com/volume-2/chapter-1/


Fix has been checked into Experimental Tab branch.


> Also, the TN notes are broken, but I guess there's no easy solution for that...

Interesting.  Looks like there's a link from the text to the footnote, but no link from footnote back to text.
I've have a least fixed the link to the footnote by changing the link href to be just have the fragment of the URL.  (Again on Experimental Tab branch)
e.g. 
```html
<a href="#PTL1"><sup>1</sup></a>
```

This was another known issue. Item 3 on https://github.com/dteviot/WebToEpub/issues/64

**Comment by IcePhantom22** [comment]
Would it be possible to add [http://www.sousetsuka.com/](url) and [https://yoraikun.wordpress.com/](url)?

**Comment by dteviot** [comment]
@IcePhantom22

> Would it be possible to add http://www.sousetsuka.com/ and https://yoraikun.wordpress.com

- Yoraikun can already be done using the Default parser.  See detailed notes below.
- sousetsuka can't be done with by Default parser in the plug-in version that's currently in the chrome store, but can be done with development branch https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode.  
If you don't want to wait for the Chrome store version to be updated, you'll need to download the development branch and install from source.  Instructions on how to do this can be found at the above link.

**More detailed notes.**


> https://yoraikun.wordpress.com

This site has multiple stories.
The table of contents for each story is easy to locate.
Sampling the first and last chapter: the content of each chapter is in a &lt;div&gt; element, with a className of "entry-content"

Steps
- Browse to web page with list of chapters.  e.g. https://yoraikun.wordpress.com/tlk-chapters/
- Click on WebToEpub
- You will get a warning message that Default parser will be used.  Click OK and WebToEpub will open.
- In the list of chapters to fetch, uncheck the chapters you don't want.  (The default parser will show every hyperlink it finds on the page.)
- Above the "Pack EPUB" button, there's a drop down that shows "&lt;body&gt;", change this to &lt;div&gt;, in the drop down to the right, change "First Found" to "Class is", then in edit control on line add "entry-content".  (Don't include the quotes)
- Correct the "Cover Image URL" value.
- Click "Pack EPUB" button

That said, yoraikun.wordpress.com seems to be using wordpress format.So, you could add a mapping for it to WordpressBaseParser.js
```javascript
parserFactory.register("yoraikun.wordpress.com", function() { return new WordpressBaseParser() });
```


> http://www.sousetsuka.com/ 


This site seems to have a single story "Death March kara Hajimaru Isekai Kyousoukyoku".
The table of contents appears to be at http://www.sousetsuka.com/p/blog-page_11.html
Sampling the first and last chapter: the content of each chapter is in a &lt;div&gt; element, with a className starting with "post-body"



Therefore, the steps to use the Default parser **from the experimental branch** are:
- Browse to web page with list of chapters.  e.g. http://www.sousetsuka.com/p/blog-page_11.html
- Click on WebToEpub
- You will get a warning message that Default parser will be used.  Click OK and WebToEpub will open.
- In the list of chapters to fetch, uncheck the chapters you don't want.  (The default parser will show every hyperlink it finds on the page.  Due to order of links, you will need to clicking the "Edit Chapter URLs" button to edit the list of links you want used.)
- Above the "Pack EPUB" button, there's a drop down that shows "&lt;body&gt;", change this to &lt;div&gt;, in the drop down to the right, change "First Found" to "Class starts with", then in edit control on line add "post-body".  (Don't include the quotes)
- Click "Pack EPUB" button

That said, sousetseka seems to be using blogspot format.So, you could add a mapping for it to BlogspotParser.js
```javascript
parserFactory.register("sousetsuka.com", function() { return new BlogspotParser() });
```

**Comment by typhoon71** [comment]
I am trying to save as epub from "https://translationchicken.com/2016/09/21/rezero-web-novel-fan-translation-table-of-contents/", but it fails on Firefox 49.0.1 and last avail addon build (0.0.0.021).
It seem sto be working fine on Chrome with ExperimentalTabMode brach.
Is some stuff not merged yet? (btw, I just started using Firefox and webtoepub again)

**Comment by dteviot** [comment]
@typhoon71 
> Is some stuff not merged yet? 

Chrome store version is currently 0.0.0.23.  Experimental branch is 0.0.0.24.  So, yes.  (See https://github.com/dteviot/WebToEpub/commits/ExperimentalTabMode for history.)
That said, 0.0.0.21 on Firefox 48.02 works for me, at least for the first two chapters.  
Note, 
- You do get at least one warning message.
- The "Save to File" dialog takes a while to come up after the progress bar finishes, as it takes a while to fetch all the images.

Please described the problem in more detail.
- Exactly what steps did you do.
- What happened, compared to what you expected to happen.

**Comment by Quaturn** [comment]
OK I will put it out there :-)
would it be possible to create parser for http://lnmtl.com/ ?

**Comment by typhoon71** [comment]
mmm, OK.
1. I go to "https://translationchicken.com/2016/09/21/rezero-web-novel-fan-translation-table-of-contents/"
2. I click on the webtoepub toolbar button
3. I get "No parser found for this URL. Default parser will be used. You will need to specify how to obtain content for each chapter." [which is OK]
4. I hit OK, so the general parser starts grabbing.
5. I select just 2 links, hit "Pack epub".
6. I instantly get the "NetworkError when attempting to fetch resource." message.
7. I click OK.
8. BAck to the main tab of the addon, the "Pack epub" button is greyed out.
9. I see a really small epub is saved (4kb). 
10. I don't see traffic on the network.

I will wait for Firefox to get a newer version and check again.

**Comment by dteviot** [comment]
@typhoon71 
> I will wait for Firefox to get a newer version and check again.

v0.0.0.23 is now available in Firefox store.
Side note, their approval process has gotten a lot faster.  (Or the auditor know recognises my plug-in.)

**Comment by dteviot** [comment]
@typhoon71 
>I instantly get the "NetworkError when attempting to fetch resource." message.

Strange.
Error message comes from Firefox, not liking the URL it has been asked to fetch before even trying to get it. I'll need you to get me more information

Set up WebToEpub to fetch the two chapters again, but before you click "Pack EPUB"
- Right click on the WebToEpub window and select "Inspect Element" on the menu that appears. (Don't select "Inspect Element with Firebug")
- On stuff that appears at bottom of screen, click on the "Console" menu item
- Click on "Pack EPUB" button
- Send me contents of the "Console" window.

Also, you might like to select the "Network" tab on the "Network" menu item, and click "Pack EPUB" again.  This will show you the network calls that are made.  (You can also do this with normal EPUB creation, you'll see progress of every file request, it's pretty awesome.)
Unfortunately, this is cleared when the "Download" dialog appears.  So may be helpful to log to a file. https://developer.mozilla.org/en-US/docs/Mozilla/Debugging/HTTP_logging.  (Note, if nothing appears in network, and it may not if Firefox is rejecting the attempt before it even tries to fetch it, there's no need to get this log.

**Comment by dteviot** [comment]
@Quaturn
> would it be possible to create parser for http://lnmtl.com/ ?

I'll see if it can be done.  From a quick inspection, looks like there's complications.

**Comment by typhoon71** [comment]
I tried to get the output of the console, but it's empty.
I do get the error but there's nothing there (btw, I don't have firebug).

I may have found something tough: if I whitelist the addon window in ublock origin [whitelist entry is "46fb073d-e454-4740-8e95-64f33addd34a.moz-extension-scheme"] the epub is created fine; this is strange because in chrome I didn't need to whitelist epubtopdf.

Also, when packing the epub with the addon whitelisted, the console comes to life, but the output disappears as soon as the epub is saved.
So I think ublock origin is blocking the network requests in Firefox; but it doesn't this in chrome (even if I have ublock origin websocket installed there).

So, should I file a bug on Ublock origin github? What kind of bug should it be?

**Comment by dteviot** [comment]
@typhoon71   
Sorry for delay in responding.  Hard Drive on my home PC failed.  Have to write this from work.  Will probably be 3 to 4 weeks before I get new machine and dev resumes.

> So, should I file a bug on Ublock origin github

Please do.

>What kind of bug should it be?

I don't know what the alternatives are.  I personally would go to https://github.com/gorhill/uBlock/issues
click on the "issue" button and fill in using using the following guidelines.
http://www.drmaciver.com/2013/09/how-to-submit-a-decent-bug-report/

(Hint, you can probably copy/paste most of your previous posts.)

**Comment by typhoon71** [comment]
OK, I know of ublock origin github, I already used it.
If  there's nothing specific to add then no problem, I'll do it (almost) now.

Sorry to ear you PC has failed, I know what it means... 
We'll wait eagerly for your return. XD

**Comment by dteviot** [comment]
@Quaturn
> would it be possible to create parser for http://lnmtl.com/ ?

Basic support has been added to the this branch. https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode
You'll need to download and install from source.  (Instructions are on the readme at above URL.)
Note, at current time, it will only find the 50 chapters that appear on the index page.  I haven't yet figured out all the REST calls that need to be made to fetch a list of all chapters.

**Comment by dteviot** [comment]
Hmmm.  A whole bunch of parsers.  https://github.com/JimmXinu/FanFicFare/tree/master/fanficfare/adapters
And all I'll need is a JavaScript interpreter for Python.

**Comment by Quaturn** [comment]
@dteviot 
First thank you very much for LNMTL it is working great as for adding more links I was able to work around it by copying the generated links to notepad++ and then using column editor

There is one thing however, would it be possible to add chapter title from web page to the chapter body?

The reason is that when generating epub with too many xhtml ereaders have problem to load the epub and when merging the file (in my case every fifty chapters) there is no chance of recognizing which chapter i am reading as many pages do not have chapter number in the body of the text

**Comment by dteviot** [comment]
@Quaturn 
> would it be possible to add chapter title from web page to the chapter body?

I'll see what I can do.  
I haven't finished working on the LNMTL parser.

**Comment by dteviot** [comment]
@Quaturn 
Have updated LNMTL on ExperimentalTabMode branch to add chapter titles.
It also makes REST calls to retrieve the full set of chapters.  (This can take 10 seconds or more.)
Finally, it makes a best guess at image (if any) to use for cover.
Please let me know how it works for you.

**Comment by Quaturn** [comment]
@dteviot 
Maan you've done it again it works beautifully/perfectly. Thank you very very much :-D

**Comment by belldandu** [comment]
@dteviot
`Hmmm.  A whole bunch of parsers.  https://github.com/JimmXinu/FanFicFare/tree/master/fanficfare/adapters
And all I'll need is a JavaScript interpreter for Python.`

;) you have one right here

**Comment by typhoon71** [comment]
Just a little bump for the "unlimitednovelfailures.mangamatters.com" parser, the last of 22 entries. ;)

In all honesty, I noticed it wasn't done... because I was "cleaning up" LNs backlog, and found myself thinking "why not making epubs of the novels from UNF myself?".

Side note: the novels is on a single html, with link to sections. I suppose the additional work would be for the illustrations, which are on another page.

They just changed theme btw.

**Comment by typhoon71** [comment]
I found a page where the parser... doesn't grab anything, producing an empty epub.
Link: "https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc"

When the empty epub is pached, webtoepub gives this error:
'Warning, content element for web page 'https://tachibanachinatsu.wixsite.com/tenshitranslations' has no visible content.'

I don't understand why this is happening, maybe a new parser is needed for wixsite.com?

**Comment by dteviot** [comment]
@typhoon71
> I found a page where the parser... doesn't grab anything, producing an empty epub.
Link: "https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc"

It happens because the HTML pages listed in the table of contents don't have any content.  Instead the javascript with each HTML page makes REST calls to get the content text and adds that into the page body.
A new parser will be needed to examine the HTML, make the required REST calls and compose the content text.   The LNMTL and GravityTales sites do something similar for the chapters making up a novel.

**Comment by dteviot** [comment]
> I see there's another unchecked entry on it, did I put it there?

I assume you mean unlimitednovelfailures.mangamatters.com.  I think it's one you requested.  It's also unlikely to ever be done, due to fact that:
* site has at least 3 different formats, 
* and most of the content can be found elsewhere
* the default parser can collect the content (if not very well.)  Possibly I should add a second list of  "sites I won't/can't make a parser for."  and move this to it.

> I don't even remember adding all those parser to the list!

You didn't.  I've added many of them as people have requested additional parsers.

> why do they make a page so complicated? Isn't it bad for the server?

See: https://www.w3schools.com/xml/ajax_intro.asp

**Comment by typhoon71** [comment]
Mmmm, I may have accidentally deleted an entry, "q...-something". 
I'll see if I can fix that, sorry.

About unlimitednovelfailures.mangamatters.com: yes, I was the one asking for it.
Right now the default parser can grab it well enough.
If you edit the chapters list you can have a perfect epub with a toc at the start; I'd say it's OK to not work on it, so you can remove the entry if you wish so (or add it to your won't do list).

**Comment by dteviot** [comment]
> Mmmm, I may have accidentally deleted an entry, "q...-something".

That was probably me, https://en.qidian.com/ (aka https://www.webnovel.com/) was done but I hadn't marked it off.

**Comment by typhoon71** [comment]
Yes, that's it, "quidian" was the one I accidentally deleted.
Good to hear there's no damage done.

**Comment by dteviot** [comment]
@typhoon71
I've put together a "proof of concept" parser for Wixsite and checked into Experimental Tab Mode.  Please give it a try and let me know how it goes.
Points to note:
* At moment, really just fetches the content for each chapter.
* Not all chapters will have a title.  (The title part seems to be in different places on different pages.)
* I've only tried a few pages from https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc, but aside from the issue with the titles, it seems to work.

**Comment by typhoon71** [comment]
Seems fine to me; I use it for https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc, the whole thing (58 chapters) and it works.
As you said, some (most of) titles aren't fetched, but at least the chapter title is present in the TOC of the epub.
If fetching the title from the page proves not feasible, you could just grab it from the TOC instead.

**Comment by dteviot** [comment]
@typhoon71
Of course I could do it that way, but that's cheating.
On the other hand, it's easy to do and there's only one story you're interested in.
So, Done.
Enjoy.

**Comment by typhoon71** [comment]
Great, thanks. And again, who said that cheating was bad? XD

**Comment by dteviot** [comment]
>who said that cheating was bad?

It doesn't work well for https://tachibanachinatsu.wixsite.com/tenshitranslations/tng-toc
Also, the image at https://tachibanachinatsu.wixsite.com/tenshitranslations/vol-1-ch-1-part-3 isn't being picked up either.

**Comment by typhoon71** [comment]
Well, that novel (The New Gate) is something I'll read later on, so not a "problem" right now. XD
Btw, I didn't even test it.
Also, one can read only a limited amount of novels, based on time mostly...

**Comment by dteviot** [comment]
At least some of "The New Gate" is on Baka Tsuki.  https://www.baka-tsuki.org/project/index.php?title=The_New_Gate
Looks like original source is: https://shintranslations.com/the-new-gate-toc/  and the plug-in has no trouble with that site.  (Not really surprising, looks like it's Wordpress based.)

**Comment by typhoon71** [comment]
Well, in the end I did investigate for sources of "The New Gate" and I'm already reading the second volume from Baka-Tsuki (made with webtoepub). There is Shin TL as a source too.
I think the stuff on wixsite may be different (WN/LN) or a duplication (sometime ppl "steal" TLs too).

**Comment by typhoon71** [comment]
I noticed that when you grab from "https://shintranslations.com/the-new-gate-toc/" the "Illustrations" page doesn't contain the full sized images, I think it's because there's yet another level/link to them.
Can you please fix it? Thanks.

**Comment by dteviot** [comment]
@typhoon71
Can you be more specific?
* Which of the illustrations pages is the problem?  (url please)
* Exactly which image(s)?  (url please)
* What are the dimensions of the image file that is collected?  (Unzip the epub and check the image file properties.)
* What are the dimensions you expect?

As far as I can tell, the full size images are obtained for Volume 1.

**Comment by typhoon71** [comment]
Oh, you're right, the image that are grabbed are indeed the full sized ones, my bad.

The issue the is that webtoepub seems to tries to "reconstruct" the page layout (the "grid" of images), scaling the images to a really smal size at view time (calibre).

It happens with all of the links for the illustrations galleries from "https://shintranslations.com/the-new-gate-toc/", for example "https://shintranslations.com/vol-3-illustrations/".

Cheching the epub I found out that there's code like this:

'<div class="gallery-row" data-original-width="728" data-original-height="265">
     <div class="gallery-group images-1" data-original-width="182" data-original-height="265">'

which I think is the cause. 

I think it would be better to have them grabbed and formatted as usual, 1 image per page.

**Comment by buthub** [comment]
Hi, can I ask for a parser to be created for this new site,
http://novelfull.com/
I like it cause it seems to include a lot of chapter titles that other sites don't bother with.  Element seems to be div id="chapter-content".

Also, for those interested, how about one for Google Translate?
https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=http%3A%2F%2Fwayback.archive.org%2Fweb%2F20130318180132%2Fhttp%3A%2F%2Fncode.syosetu.com%2Fn4353bc%2F

https://translate.google.com/translate?hl=en&sl=ja&tl=en&u=http%3A%2F%2Fncode.syosetu.com%2Fn3009bk%2F&sandbox=1

**Comment by Maradar** [comment]
Hello,

Can I ask for parser for this page:
http://liberspark.com

Thanks

**Comment by dteviot** [comment]
@Maradar
Have added to branch https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode, if you'd like to load from source and try it.

**Comment by Maradar** [comment]
@dteviot 
Thanks it works :-D

**Comment by KiraYamatoSD** [comment]
Hello,

Can I ask for parser for this page:
http://www.wuxiaworld.co/

The site looks similar to http://novelfull.com/

Thanks

**Comment by dteviot** [comment]
@KiraYamatoSD 
I have added to branch https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode, if you'd like to load from source and try it. (See https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode for instructions for loading from source.)
If you need additional parsers, please read this first:
https://dteviot.github.io/Projects/webToEpub_CustomizingParserTemplate.html

**Comment by ccrowles** [comment]
Hi, if it's not too much of a bother can I ask for a parser for this site?
https://xuxunette.wixsite.com/danmeitranslations/faraway-wanderers

Thanks in advance and sorry for bothering you.

**Comment by dteviot** [comment]
@ccrowles,
Sorry, I missed this request.  It's been years since this issue has been updated.
It's better to create a new Issue for each site.
Makes traceability easier.

**Comment by dteviot** [comment]
@ccrowles,
I had a quick look (30 minutes) at faraway-wanderers.
But I'm sorry to say, it's not an easy one to do.  
As you might guess, as there's a request for https://tachibanachinatsu.wixsite.com/tenshitranslations/copy-of-obc-toc at the top of this issue, from 6 years ago that I still have not done.

**Comment by ccrowles** [comment]
I see, thanks anyways!
